scene/gui/text_edit.cpp

```diff
@@ -1108,6 +1108,7 @@ void TextEdit::_notification(int p_what) {
 						if (cursor.column == last_wrap_column + j && cursor.line == line && cursor_wrap_index == line_wrap_index) {
 
 							cursor_pos = Point2i(char_ofs + char_margin + ofs_x, ofs_y);
+							cursor_pos.y += (get_row_height() - cache.font->get_height()) / 2;
 
 							if (insert_mode) {
 								cursor_pos.y += (get_row_height() - 3);
```

Comment:
For `insert_mode` the following should do the trick:

```suggestion
 								cursor_pos.y += (cache.font->get_height() - 3);
```

It will have to also be applied [here.](https://github.com/godotengine/godot/blob/c570c4300765e1fc85d4492590c62e8fa71d8245/scene/gui/text_edit.cpp#L1206)

Reply:
Done. Also rebased the commits into one.